### ISSUE TITLE
2D X-ray image

### ISSUE THREAD

**Issue by 9B8DY6** [fepegar/torchio#358]
Hi! I am the one who asked about how to use torchio in 2D xray image segmentation. Thank you so much for giving me answer about the previous problem! I don't know how to reopen so I asked in this way.

I have another question.
Is it able to do torchvision transforms into torchio.dataset images?

When I did torchio.transform to 2D xray image dataset, CroporPad is only for an integer or tuple of 3, not (256,256).
![image](https://user-images.githubusercontent.com/67573223/99664135-11b85780-2aab-11eb-90e5-a400d28b7dba.png)
I did transforms to training set like that but the result was that the image shape is (3,256,256,1) and the image didn't come out.  How can i solve it? rgb_to_gray didn't work...
![CHNCXR_0002_0](https://user-images.githubusercontent.com/67573223/99665746-65c43b80-2aad-11eb-978e-ada9ff0ea27c.png)
![CHNCXR_0001_0](https://user-images.githubusercontent.com/67573223/99665778-6d83e000-2aad-11eb-88c3-36f50ad95c07.png)
![CHNCXR_0002_0_mask](https://user-images.githubusercontent.com/67573223/99665782-6eb50d00-2aad-11eb-8dc9-2ee6785e99c4.png)
![CHNCXR_0001_0_mask](https://user-images.githubusercontent.com/67573223/99665786-6eb50d00-2aad-11eb-909d-5670cc584dfb.png)
[Code&csv.zip](https://github.com/fepegar/torchio/files/5566686/Code.csv.zip)

**Comment by fepegar** [comment]
Hi again! There should be a "Reopen issue" at the bottom of #356.

> Is it able to do torchvision transforms into torchio.dataset images?

I'm not sure what you mean. Can you provide some code? I think `torchvision` transforms take 2D tensors. I suppose you could pass the tensor data within the image (`image.data[..., 0]`).

> When I did torchio.transform to 2D xray image dataset, CroporPad is only for an integer or tuple of 3, not (256,256).

As the docs say, you need to pass a tuple of three values to `CropOrPad`. In your case, you can use `(256, 256, 1)`.

**Comment by fepegar** [comment]
Thanks for sharing the code. It would help a lot if you could share a minimal working example, pasting actual code (instead of a screenshot of it) and the image you're using, so that I can investigate the issue properly.

**Comment by 9B8DY6** [comment]
I upload two xray png, each mask and Zip file of code. As order, first xray image matches first mask.

**Comment by fepegar** [comment]
Could you please also provide a script I can run?

**Comment by 9B8DY6** [comment]
[영상처리_1.zip](https://github.com/fepegar/torchio/files/5566761/_1.zip)
Is it right to upload .py file...? I am really sorry that I don't know the meaning of script...

**Comment by fepegar** [comment]
Thanks. A script is just a text file containing a Python program. It's similar to what you've shared, which is a script generated from a Jupyter Notebook. I cannot run it because I don't have access to your Google Drive. If your only concern is that the final image has 3 channels... it's my fault! The indexing in `rgb_to_gray` should be `[:1]` to get only the first channel keeping the shape, not `[0:]`, which does nothing. Sorry about that! let me know if that helped.

**Comment by fepegar** [comment]
Feel free to reopen (using the button below) if you have more questions.
